tests/align/test_align_bugs.cpp

```diff
@@ -0,0 +1,111 @@
+// ==========================================================================
+
+#include <seqan/basic.h>
+#include <seqan/stream.h>
+
+#include <seqan/align.h>
+
+// Call the internal interface since we cannot publicly control the Traceback configuration.
+template <typename TGapSequence1, typename TGapSequence2, typename TScoringScheme>
+auto compute_affine_alignment(TGapSequence1 & gapSequence1,
+                              TGapSequence2 & gapSequence2,
+                              TScoringScheme const & scoringScheme)
+{
+    using TSize = typename seqan::Size<TGapSequence1>::Type;
+    using TPosition = typename seqan::Position<TGapSequence1>::Type;
+    using TTraceSegment = seqan::TraceSegment_<TPosition, TSize>;
+
+    // Test explicitly the complete trace functionality.
+    using TTracebackPolicy = seqan::TracebackOn<seqan::TracebackConfig_<seqan::CompleteTrace, seqan::GapsLeft>>;
+    using TAlignConfig = seqan::AlignConfig2<seqan::LocalAlignment_<>,
+                                             seqan::DPBandConfig<seqan::BandOff>,
+                                             seqan::FreeEndGaps_<seqan::True, seqan::True, seqan::True, seqan::True>,
+                                             TTracebackPolicy>;
+
+    seqan::String<TTraceSegment> trace;
+    seqan::DPScoutState_<seqan::Default> dpScoutState;
+    TAlignConfig config;
+    auto score = seqan::_setUpAndRunAlignment(trace,
+                                              dpScoutState,
+                                              seqan::source(gapSequence1),
+                                              seqan::source(gapSequence2),
+                                              scoringScheme,
+                                              config,
+                                              seqan::AffineGaps());
+    seqan::_adaptTraceSegmentsTo(gapSequence1, gapSequence2, trace);
+    return score;
+}
+
+SEQAN_DEFINE_TEST(test_local_alignment_with_complete_trace_produces_wrong_alignment)
+{
+    seqan::String<seqan::AminoAcid> s1 = "QDPKEPCISIYCLYLPVAGLGNLRGCCLPWM**";
+    seqan::String<seqan::AminoAcid> s2 = "*SS*IVSRTPARTSPPGPGPIGGPH*TIQSVVATVGVYKGQGRNQRELMTRAYWEFLVQGK*LQFPIPST"
+                                         "TGVQRVSRTCRPRRAHADPVSVARVRPRTSKGITDLLLLNLVWLNATCPSKKLNADRDGRVTI*QARVSF"
+                                         "VIGINQTNRSTN*ERPCTTTHRIKKELSICQSSLCPGRVRFPVLSQIKPQAPLLVVPFRQFL*VSALQPY"
+                                         "FPRNPKTLVSRKLPEGSSM*RPPIASWHRL*SELGRYLIVFEPLTFALD*RKHSWQMLSQSFVLRRSKNF"
+                                         "TSNGTVRIAPVCPS*SLPRAPKTNKIEPRSYSIIPCTIIQAREPALNTLIFSK*TFRPPPTLSQEHRRRT"
+                                         "GGKARTSSTRLATDRPPAPKIQLRAF*PQQLKYILLELELPRLLAPDLPSNGYSLKLLKCTHSNYRASNE"
+                                         "SCIVIFRHYLPASGMGNLRACCLPWMW*PFLRLPLRNRTLIPRHP*EPR*ANTVPTKVDRADT*MIRRRC"
+                                         "*TVRSAKLSRVTKANAPEDAAGFWSDKCT";
```

Comment:
Isn't there a much more minimal example? I don't understand what should be right and what is wrong with this example.

Reply:
This was a reported bug and the offending code which I used to test and fix.